File: heart/src/main/java/com/vantagecircle/heartrate/fragment/presenter/HistoryFragmentPresenter.java
package com.vantagecircle.heartrate.fragment.presenter;

import android.content.Context;

import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.vantagecircle.heartrate.adapter.HistoryAdapter;
import com.vantagecircle.heartrate.data.DataManager;
import com.vantagecircle.heartrate.model.HistoryModel;

import java.util.ArrayList;

/**
 * Created by bapidas on 08/11/17.
 */

public class HistoryFragmentPresenter {
    private static final String TAG = HistoryFragmentPresenter.class.getSimpleName();
    private DataManager mDataManger;
    private Context mContext;
    private RecyclerView mRecyclerView;

    public HistoryFragmentPresenter(Context mContext, DataManager mDataManger, RecyclerView mRecyclerView) {
        this.mDataManger = mDataManger;
        this.mContext = mContext;
        this.mRecyclerView = mRecyclerView;
    }

    public void initialize() {
        LinearLayoutManager mLinearLayoutManager = new LinearLayoutManager(mContext);
        mLinearLayoutManager.setOrientation(LinearLayoutManager.VERTICAL);
        mRecyclerView.setLayoutManager(mLinearLayoutManager);
        mRecyclerView.scrollToPosition(0);
        ArrayList<HistoryModel> mArrayList = mDataManger.getHistory();
        if (mArrayList != null && mArrayList.size() > 0) {
            HistoryAdapter mHistoryAdapter = new HistoryAdapter(mArrayList);
            mRecyclerView.setAdapter(mHistoryAdapter);
        }
    }
}

File: heart/src/main/java/com/bapidas/heartrate/ui/fragment/presenter/HistoryFragmentPresenter.kt
package com.bapidas.heartrate.ui.fragment.presenter

import androidx.fragment.app.FragmentActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.bapidas.heartrate.data.DataManager
import com.bapidas.heartrate.ui.adapter.HistoryAdapter

/**
 * Created by bapidas on 08/11/17.
 */
class HistoryFragmentPresenter(
    private val mContext: FragmentActivity?,
    private val mDataManger: DataManager,
    private val mRecyclerView: RecyclerView?
) {
    fun initialize() {
        val mLinearLayoutManager = LinearLayoutManager(mContext)
        mLinearLayoutManager.orientation = LinearLayoutManager.VERTICAL
        mRecyclerView?.layoutManager = mLinearLayoutManager
        mRecyclerView?.scrollToPosition(0)
        val mArrayList = mDataManger.history
        if (mArrayList.size > 0) {
            val mHistoryAdapter = HistoryAdapter(mArrayList)
            mRecyclerView?.adapter = mHistoryAdapter
        }
    }

    companion object {
        private val TAG = HistoryFragmentPresenter::class.java.simpleName
    }

}